src/items.c

```diff
@@ -2831,3 +2859,249 @@ void HideSpawnPoints()
 	}
 }
 
+static void SetTiming(int item_class, float seconds, float lcl, float ucl)
+{
+	if (seconds == 0)
```

Comment:
should we not check here for index out or range?
`if (item_class < NUM_ITEM_CLASSES)` is what I mean.

I know that SetTiming() is called with those enums, but still.

And for 'deathmatch' it is a must, as it can be anything...

Reply:
The reason I don't check deathmatch is because it's already bound to the range [0,5] in several other places... see FixRules() and ChangeDM()

As far as item_class goes, SetTiming is just a static helper function called only from TimingControl... We can add an assert for debug modes if you'd like but don't waste CPU cycles for a a programmer error which should be resolved through testing

Reply:
ok